src/Compute/Compute/Images/GetAzureVMImageCommand.cs

```diff
@@ -95,7 +102,69 @@ public override void ExecuteCmdlet()
 
             ExecuteClientAction(() =>
             {
-                if (this.ParameterSetName.Equals(ListVMImageParamSetName) || WildcardPattern.ContainsWildcardCharacters(Version))
+                if ((this.IsParameterBound(c => c.EdgeZone)) && this.EdgeZone != null)
+                {
+                    if (this.ParameterSetName.Equals(ListVMImageParamSetName) || WildcardPattern.ContainsWildcardCharacters(Version))
+                    {
```

Comment:
correct me if im wrong, but i think this if statement should be outside of if edge zone if statement, right?
What is customer wants to get ListofVMImage but without using edgezone?

Reply:
never mind i see what you did at the bottom